code/modules/mob/living/carbon/human/human_movement.dm

```diff
@@ -6,6 +6,11 @@
 
 	if (istype(loc, /turf/space)) return -1 // It's hard to be slowed down in space by... anything
 
+	if (istype(loc, /turf/simulated/open)) //Open space things
+		if(!(locate(/obj/structure/lattice, loc) || locate(/obj/structure/stairs, loc) || locate(/obj/structure/ladder, loc)))
+			if(!has_gravity(src, loc))

```

Comment:
Move this to the clause at line 9. It's cheaper to check for this, than it is to use the locate calls.

`if (isopenturf(loc) && !has_gravity(src, loc))`